lib/bambora/client.rb

```diff
@@ -1,47 +1,107 @@
 # frozen_string_literal: true
 
-require 'base64'
-require 'faraday'
+require 'bambora/client/version'
+require 'bambora/headers'
+require 'bambora/rest_client'
+require 'bambora/json_client'
+require 'bambora/v1/batch_payment_report_resource'
+require 'bambora/v1/batch_payment_resource'
+require 'bambora/v1/payment_resource'
+require 'bambora/v1/profile_resource'
 
 module Bambora
   class Client
-    attr_accessor :base_url, :merchant_id, :sub_merchant_id
+    class Error < StandardError; end
+    class ServerError < StandardError; end
 
+    attr_accessor :base_url, :merchant_id, :sub_merchant_id, :profiles_api_key, :payments_api_key, :batch_api_key,
+                  :reporting_api_key
+
+    # Initialze a new Bambora::Client.
+    #
+    # @example
+    #
+    #   client = Bambora::Client.new do |c|
+    #     c.base_url = ENV.fetch('BAMBORA_BASE_URL')
+    #     c.merchant_id = ENV.fetch('BAMBORA_MERCHANT_ID')
+    #     c.sub_merchant_id = ENV.fetch('BAMBORA_SUB_MERCHANT_ID')
+    #     c.profiles_api_key = ENV.fetch('BAMBORA_PROFILES_API_KEY')
+    #     c.payments_api_key = ENV.fetch('BAMBORA_PAYMENTS_API_KEY')
+    #     c.batch_api_key = ENV.fetch('BAMBORA_BATCH_API_KEY')
+    #     c.reporting_api_key = ENV.fetch('BAMBORA_REPORTING_API_KEY')
+    #   end
+    #
+    # @param options[base_url] [String] Bambora Base URL
+    # @param options[merchant_id] [String] The Merchant ID for this request.
+    # @param options[sub_merchant_id] [String] The Sub-Merchant ID for this request.
+    # @param options[profiles_api_key] [String] The Profiles API Key for this request.
+    # @param options[payments_api_key] [String] The Payments API Key for this request.
+    # @param options[batch_api_key] [String] The Batch API Key for this request.
+    # @param options[reporting_api_key] [String] The Reporting API Key for this request.
     def initialize(options = {})
-      unless options[:version].nil?
-        raise Bambora::Error, 'Only V1 endpoints are supported at this time.' if options[:version].upcase != 'V1'
+      if !options[:version].nil? && options[:version].upcase != 'V1'
```

Comment:
This condition is a little confusing. Maybe it could be easier to read with like ` if options[:version].present?`

Reply:
`.present?` only exists in Rails! Writing Ruby without Rails makes you appreciate all the nice things Rails does for you.